paper/paper.md

```diff
@@ -21,158 +21,63 @@ authors:
-- uniqueness
-- allowed values
-
-Compared to `datapact`, it has a small number of supported constraints, does not output any human-readable documentation,
-and does not integrate well with Jupyter Notebooks.
-It does, however, have a data profiler that seems useful for setting up an initial expectation suite.
-
-## Standalone Services
-
-The following tools all work very similar:
-They are commercial Software-as-a-Service offerings that connect directly to data warehouses (all support Snowflake, Redshift and BigQuery), do semi-automatic data monitoring and tracking, and alert via Slack or Email when
-they find an anomaly.
-
-Typically geared more towards industry applications than research, their common pitch is "being datadog for data", and they aim to prevent data pipelines from breaking.
-Compared to `datapact`, they are easier to set up, but very limited in the types of errors they can catch.
-Documentation is not in scope for them.
-
-### Monte Carlo
-
-Monte Carlo (https://www.montecarlodata.com) describes itself with the tagline "Data Reliability Delivered.".
-It monitors freshness, volume and schema changes to all tables.
-Anomaly detection is performed using machine learning, which can make it hard to predict how the tool works.
-For usecases where SaaS-usage is prohibited, e.g. because of data privacy requirements,
-hybrid deployments are only supported on AWS, ruling Monte Carlo out for a lot of research usecases.
-
-Monte Carlo features ready-made integrations for orchestrators like AirFlow, enabling custom-made
-workflows e.g. for verifying pull requests.
-
-### Metaplane
-
-Metaplane (https://metaplane.dev) has the literal tagline "The Datadog for data".
-
-Similar to Monte Carlo, it connects to a data warehouse and performs anomaly detection
-on metrics like row and column count, data freshness, uniqueness, distribution, or custom SQL-based metrics.
-Anomalies are alerted via the Slack integration, and allow data engineers to give model feedback to train it better.
-
-Metaplane, like Montecarlo, is not available for on-premise hosting.
+To ensure high code quality, `datapact` maintains 100% code coverage via unit tests, automatically executed by our continuous integration.
+Code style is linted via pylint (@pylint), pyright (@pyright) and Prettier (@prettier).
+Documentation is auto-generated from docstrings (@goodger2010docstring).
 
-### BigEye
+# Example
 
-In addition to the common data ware houses, BigEye also connects to systems like Presto, SAP, Oracle and Rockset.
-Apart from data freshness, volume (row number, nullability, uniqueness ...) and summary statistics, it can
-verify a column's data format (UUID, zip codes, ...).
-It tracks data freshness, volume (), format (UUID, zip codes, ...) and summary
-statistics, and performs basic outlier detection.
-Compared to the other tools, BigEye seems to be the most lacking feature-wise.
+![Statistical assertions on the Iris dataset.\label{fig:example}](screenshot.png)
```

Comment:
In my last submission one of the reviewers asked me to create a minted environment for my code example. I would like to suggest a minted environment here as well, since it still displays nicely and is easier for readers to copy/paste rather than typing what is on the image. The results can still be images, but I would suggest adding a vector image, e.g., pdf or eps so it doesn't look pixelated on high resolution screens

Reply:
could you elaborate on what a minted image is?

this is a screenshot from VS Code, I don't see an easy way of making it a vector graphic without rebuilding it in a drawing program; which defeats the purpose of a screenshot. The image is already pretty high-resolution, but I could try upsampling it artificially if that helps.

Reply:
About the minted environment, it automatically highlights code for you and it is still text that can be copy/pasted. For example [https://pt.overleaf.com/learn/latex/Code_Highlighting_with_minted](https://pt.overleaf.com/learn/latex/Code_Highlighting_with_minted).

The stuff after the code can be an image, but a screenshot might not produce the best quality. If you are using a jupyter notebook, you can open it on chrome and go to `Developer Tool > Rendering > Select 'screen' in Emulate CSS media`. Then you should be able to print a PDF with the screen CSS and it will be in vector format already. I guess you can also include the code in this vector image if the minted environment is too much effort.

Reply:
Oh, you mean a code block! Good to know that LaTeX calls that „minted“ :)

When using a codeblock, I think it wouldn‘t be obvious that this is from a Jupyter Notebook. I might be thinking to pragmatic here, a screenshot of the current resolution sounds like a fine solution to me.

Reply:
I just checked and it was better only with the image since it is redundant now. Don't worry about it, unless the reviewers ask for the code block.

Reply:
okidoke 👍 reverted the change in https://github.com/Skn0tt/datapact/pull/183/commits/caf3fa066b03911b7d7a3cf837c4516b22788162.